### PR TITLE
Add search and filters to Pages list page

### PR DESCRIPTION
**Pull request by Cloud11PL**
I want to merge this change because it:
- Adds search to pages list page
- Adds "page type" filter to pages list page
- Updates storybook

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3516 Add search to pages](https://app.clickup.com/t/2549495/SALEOR-3516)

File: src/pages/components/PageListPage/PageListSearchAndFilters.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        onSearchChange={handleSearchChange}
+        searchPlaceholder={intl.formatMessage(messages.searchPlaceholder)}
+        allTabLabel={"All Pages"}
+        tabs={tabs.map(({ name }) => name)}
+        currentTab={currentTab}
+        onTabDelete={handleTabDelete}
+        onTabChange={handleTabChange}
+        onTabSave={() => openModal("save-search")}
+      />
+      <SaveFilterTabDialog
```
Comment (dominik-zeglen) [inline src/pages/components/PageListPage/PageListSearchAndFilters.tsx:121]: Its not consistent with the rest of the project, where we keep dialog outside filtr component.

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

File: src/pages/components/PageListPage/PageListSearchAndFilters.tsx (hunk 1 above)
Comment (Cloud11PL) [reply src/pages/components/PageListPage/PageListSearchAndFilters.tsx:121]: @dominik-zeglen Gift Cards has been done like that - https://github.com/saleor/saleor-dashboard/blob/main/src/giftCards/GiftCardsList/GiftCardListSearchAndFilters/GiftCardListSearchAndFilters.tsx

**Review by orzechdev — APPROVED** [review]

File: src/pages/components/PageListPage/PageListSearchAndFilters.tsx (hunk 1 above)
Comment (dominik-zeglen) [reply src/pages/components/PageListPage/PageListSearchAndFilters.tsx:121]: Yes, inconsistency is the downside. Patterns used in gift card development were experimental and we did not reflect on this yet.

**Review by dominik-zeglen — APPROVED** [review]

**Pull request merged**